src/libdrakvuf/win-processes.c

```diff
@@ -139,9 +139,14 @@ addr_t win_get_current_thread(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
     addr_t thread = 0;
     addr_t prcb = 0;
     addr_t kpcr = 0;
+    uint64_t ss_arbytes = 0;
 
-    // TODO: check whether we could use ss_arbytes here
-    unsigned int cpl = info->regs->cs_sel & 3;
+    if (VMI_FAILURE == vmi_get_vcpureg(vmi, &ss_arbytes, CS_SEL, info->vcpu))
```

Comment:
It would be best to avoid using `vmi_get_vcpureg` here because it's an expensive extra hypercall. We would need to use what's provided in `info->regs`

Reply:
`cs_sel` and `ss_arbytes` have zero values each time. I think this is sort of optimization. Though I didn't check Xen sources yet. But this works for me.

Reply:
Thank you very much. Copy&Past error. Though I should say that in all tests CS selector works. But due documentation SS AR is rocks. I will fix it.